packages/circuits/jwt-verifier-template.circom

```diff
@@ -0,0 +1,327 @@
+    signal azpKeyMatch[azpKeyLength] <== RevealSubstring(maxPayloadLength, azpKeyLength, 1)(payload, azpKeyStartIndex, azpKeyLength);
+    for (var i = 0; i < azpKeyLength; i++) {
+        azpKeyMatch[i] === azpKey[i];
+    }
+
+    // Reveal the azp in the payload
+    signal azpStartIndex <== azpKeyStartIndex + azpKeyLength + 1;
+    signal azpMatch[maxAzpLength] <== RevealSubstring(maxPayloadLength, maxAzpLength, 0)(payload, azpStartIndex, azpLength);
+    azp <== Bytes2Ints(maxAzpLength)(azpMatch);
+
+    // Verify if the key `email` in the payload is unique
+    var emailKeyLength = EMAIL_KEY_LENGTH();
+    var emailKey[emailKeyLength] = EMAIL_KEY();
+    signal emailKeyMatch[emailKeyLength] <== RevealSubstring(maxPayloadLength, emailKeyLength, 1)(payload, emailKeyStartIndex, emailKeyLength);
+    for (var i = 0; i < emailKeyLength; i++) {
+        emailKeyMatch[i] === emailKey[i];
+    }
+
+    // Reveal the email in the payload
+    var maxEmailLength = EMAIL_ADDR_MAX_BYTES();
+    signal emailStartIndex <== emailKeyStartIndex + emailKeyLength + 1;
+    signal email[maxEmailLength] <== RevealSubstring(maxPayloadLength, maxEmailLength, 0)(payload, emailStartIndex, emailLength);
+
+    // Calculate account salt using email
+    var numEmailInts = compute_ints_size(maxEmailLength);
+    signal emailInts[numEmailInts] <== Bytes2Ints(maxEmailLength)(email);
+    accountSalt <== AccountSalt(numEmailInts)(emailInts, accountCode);
+
+    // Verify if the key `nonce` in the payload is unique
+    var nonceKeyLength = NONCE_LENGTH();
+    var nonceKey[nonceKeyLength] = NONCE();
+    signal nonceKeyMatch[nonceKeyLength] <== RevealSubstring(maxPayloadLength, nonceKeyLength, 1)(payload, nonceKeyStartIndex, nonceKeyLength);
+    for (var i = 0; i < nonceKeyLength; i++) {
+        nonceKeyMatch[i] === nonceKey[i];
+    }
+
+    // Reveal the command in the nonce
+    signal commandStartIndex <== nonceKeyStartIndex + nonceKeyLength + 1;
+    signal command[maxCommandLength] <== RevealSubstring(maxPayloadLength, maxCommandLength, 0)(payload, commandStartIndex, commandLength);
+
+    // Check if the command in the nonce has a valid invitation code and remove the prefix if it exists
+    signal prefixedCodeRegexOut, prefixedCodeRegexReveal[maxCommandLength];
+    (prefixedCodeRegexOut, prefixedCodeRegexReveal) <== InvitationCodeWithPrefixRegex(maxCommandLength)(command);
+    isCodeExist <== IsZero()(prefixedCodeRegexOut-1);
+    signal removedCode[maxCommandLength];
+    for(var i = 0; i < maxCommandLength; i++) {
+        removedCode[i] <== isCodeExist * prefixedCodeRegexReveal[i];
+    }
+
+    // Check if the command in the nonce has a valid email address and remove the email address if it exists
+    signal emailAddrRegexOut, emailAddrRegexReveal[maxCommandLength];
+    (emailAddrRegexOut, emailAddrRegexReveal) <== EmailAddrRegex(maxCommandLength)(command);
+    signal isEmailAddrExist <== IsZero()(emailAddrRegexOut-1);
```

Comment:
This can be simplified to `isEmailAddrExist <== emailAddrRegexOut` as emailAddrRegexOut is either 0 or 1.